src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Console/PluginsCli.cs

```diff
@@ -0,0 +1,185 @@
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError("Unhandled exception occurred: {0}", ex.Message);
+            }
+
+            return 1;
+        }
+        private int RunPack(PackOptions packOptions)
+        {
+            if (!TryGetValidatedPackArgs(packOptions, out PackArgs? packArgs))
+            {
+                return 1;
+            }
+
+            return this.packCommand.Run(packArgs);
+        }
+
+        private int RunMetadataGen(MetadataGenOptions metadataGenOptions)
+        {
+            if (!TryGetValidatdMetadataGenArgs(metadataGenOptions, out MetadataGenArgs? metadataGenArgs))
+            {
+                return 1;
+            }
+
+            return this.metadataGenCommand.Run(metadataGenArgs);
+        }
+
+
+        private int HandleParseError(IEnumerable<Error> errs)
+        {
+            string errors = string.Join(Environment.NewLine, errs.Select(x => x.ToString()));
+            this.logger.LogError($"Failed to parse command line arguments: \n{errors}");
+
+            return 1;
+        }
+
+        private bool TryGetValidatedPackArgs(PackOptions options, out PackArgs? result)
+        {
+            result = null;
+            if (!TryGetValidatedCommonArgs(options, out PackGenCommonArgs resultBase))
+            {
+                return false;
+            }
+
+            if (options.OutputFilePath == null && options.Overwrite)
+            {
+                this.logger.LogError("Cannot overwrite output file when output file is not specified.");
+                return false;
+            }
+
+            string? outputFileFullPath = null;
+            if (options.OutputFilePath != null)
+            {
+                if (!Path.GetExtension(options.OutputFilePath).Equals(PackageConstants.PluginPackageExtension, StringComparison.OrdinalIgnoreCase))
+                {
+                    this.logger.LogError($"Output file must have extension '{PackageConstants.PluginPackageExtension}'.");
+                    return false;
+                }
+
+                try
+                {
+                    outputFileFullPath = Path.GetFullPath(options.OutputFilePath);
+                }
+                catch (Exception ex)
+                {
+                    this.logger.LogError("Unable to get full path to output file.", ex);
+                    return false;
+                }
+
+                string? outputDir = Path.GetDirectoryName(outputFileFullPath);
+                if (!Directory.Exists(outputDir))
+                {
+                    this.logger.LogError($"The directory '{outputDir}' does not exist. Please provide a valid directory path or create the directory and try again.");
+                }
+            }
+
+            result = new PackArgs(resultBase, outputFileFullPath);
+            return true;
+        }
+
+        private bool TryGetValidatdMetadataGenArgs(MetadataGenOptions options, out MetadataGenArgs? result)
```

Comment:
I think the `TryGetValidatedXArgs` should all happen in different files. They could happen inside the ICommands themselves. I don't think this class needs to have any specific knowledge of specific options, though

Reply:
I think they should happen outside of ICommands because ICommands are not supposed to understand XOptions (cli framework dependent types) directly. These methods are transforming the cli specific options to "app" args while also validating these cli inputs. It makes sense to validate them in ICommands but not transform them. Any suggestions?

Reply:
You can have an `IOptionsValidator<TOptions, TArgs>` that the commands depend on?

Reply:
This way you can also unit test the validation logic in a single class